pkg/api/types.go

```diff
@@ -802,6 +1235,11 @@ type EnvVar struct {
 	Name string `json:"name"`
 	// Value of the environment variable.
 	Value string `json:"value,omitempty"`
+
+	// NOT YET SUPPORTED, ADDED FOR CONFORMANCE
+	// Optional: Specifies a source the value of this var should come from.
+	// +optional
+	ValueFrom *core.EnvVarSource
```

Comment:
I think we can leave this one commented out for now, it'll take a while to implement it.